dot/build_spec_test.go

```diff
@@ -16,136 +16,193 @@
 package dot
 
 import (
-	"encoding/json"
-	"fmt"
-	"io/ioutil"
+	"errors"
 	"os"
 	"testing"
 
 	"github.com/ChainSafe/gossamer/lib/genesis"
-	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/assert"
 )
 
-func TestBuildFromGenesis(t *testing.T) {
-	file, err := genesis.CreateTestGenesisJSONFile(false)
-	defer os.Remove(file)
-	require.NoError(t, err)
-	bs, err := BuildFromGenesis(file, 0)
-
-	expectedChainType := "TESTCHAINTYPE"
-	expectedProperties := map[string]interface{}{
-		"ss58Format":    0.0,
-		"tokenDecimals": 0.0,
-		"tokenSymbol":   "TEST",
+func TestBuildSpec_ToJSON(t *testing.T) {
+	type fields struct {
+		genesis *genesis.Genesis
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   []byte
+		err    error
+	}{
+		{
+			name:   "normal conditions",
+			fields: fields{genesis: &genesis.Genesis{Name: "test"}},
```

Comment:
can you add more tests to test the other attributes of `genesis.Genesis`

Reply:
added.